mantidproject/mantid
# FindCenterOfMass performace

There was a performance regression in `FindCenterOfMass` v2 which this addresses. The main cause was getting a handle to `SpectrumInfo` inside of a loop. Other renames and a healthy number of `const` has been added as well.

**To test:**

Try out the original issue and see that it is much better.

Fixes #32493

*This does not require release notes* because it fixes a performance regression introduced during the development cycle.



---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
